packages/blocks/src/api/parser.js

```diff
@@ -429,6 +429,14 @@ export function createBlockWithFallback( blockNode ) {
 		name = 'core/paragraph';
 	}
 
+	// Convert derivative blocks such as 'core/social-link-wordpress' to the
```

Comment:
Yes, this should be enough for now. In the long term, we should consider a filter that allows registering those handlers.  

```js
// Convert derivative blocks such as 'core/social-link-wordpress' to the
// canonical form 'core/social-link'.
const convertLegacySocialLink = ( { name, attributes } ) => {
    if ( name && name.indexOf( 'core/social-link-' ) === 0 ) {
        // Capture `social-link-wordpress` into `{"site":"wordpress"}`
        attributes.site = name.substring( 17 );
        name = 'core/social-link';
    }
    return { name, attributes };
}
```

You would register this type of filter just before the block gets registered.